### PR TITLE
Add dependent client repos to foreman-client-release

### PR DESCRIPTION
**Pull request by ehelms**
For plugin updates, please indicate which repos this should be built into:

* [x] Nightly
* [ ] 1.19
* [ ] 1.18
* [ ] 1.17

See Foreman's [plugin maintainer documentation](https://projects.theforeman.org/projects/foreman/wiki/How_to_Create_a_Plugin#Release-strategies) for more information.

---

### CODE REVIEW

File: packages/foreman/foreman-release/foreman-release.spec [hunk 1]
```diff
@@ -66,6 +81,18 @@ install -Dpm0644 %{SOURCE1} %{buildroot}%{repo_dir}/foreman-plugins.repo
 install -Dpm0644 %{SOURCE3} %{buildroot}%{repo_dir}/foreman-rails.repo
 install -Dpm0644 %{SOURCE5} %{buildroot}%{repo_dir}/foreman-client.repo
 
+%if 0%{?rhel}
+install -m 644 %{SOURCE6} %{buildroot}%{repo_dir}/pulp.repo
```
Comment (ekohl) [inline packages/foreman/foreman-release/foreman-release.spec]: This should be `%{SOURCE8}`

**Comment by ekohl** [comment]
I can see us using the client repo on a server install. Then you install the stable here but katello itself might want to install an explicit 2.17 for example. I'd prefer to depend on a pulp-release without a version constraint. It's more flexible and avoids the duplication.

Also thinking that the client repo can house hammer and @timogoebel also had a client tool in mind. You don't want to enable pulp repositories on all your clients IMHO.

**Comment by ehelms** [comment]
Got a suggestion then for how to handle these situations? Other ideas I had were:

 1) Pure documentation, less easy for users but avoids all issues raised
 2) Sub-package to capture these dependencies: foreman-client-release-agent, foreman-client-release-katello, foreman-client-release-gofer

**Comment by ekohl** [comment]
I'd prefer having a `pulp-release` package. IMHO we can start with hard depending on it. I think we should remove this and also depend on `pulp-release` in katello-repos`, probably with a `>= 2.17` or similar:
https://github.com/theforeman/foreman-packaging/blob/edf431b1a9e0c5bcd3ed5b3a04e3eb95a94a2d76/packages/katello/katello-repos/katello.repo#L10-L15 
https://github.com/theforeman/foreman-packaging/blob/edf431b1a9e0c5bcd3ed5b3a04e3eb95a94a2d76/packages/katello/katello-repos/katello.repo#L36-L41

It does make me wonder how we should bootstrap this in our installation instructions.

**Comment by ehelms** [comment]
That only solves the Pulp issue. Your concerns seemed to also be in general around configuring additional repositories which are also:

 * EL6: qpid-copr.repo
 * EL5: subscription-manager.repo

How would we handle this?

**Comment by ehelms** [comment]
I am leaning towards starting with updating documentation, dropping this change and updating Forklift roles to handle automated setup of these repositories when agent is set to true.

**Comment by ekohl** [comment]
Personally I don't care as much about EL5 and EL6. In the context of servers those are irrelevant. I don't mind less pretty solutions for legacy :)

Starting with documentation is also a good enough solution. What do you think of doing it in the `%desc` section as well?

**Comment by ehelms** [comment]
Oh! That comment reminded me -- EL7 is the *easy* case. Due to changes in katello-host-tools, it does not require *any* special repositories which include Pulp. Only EL6 and EL5 require special configured repositories. I have updated to try to capture that.

File: packages/foreman/foreman-release/foreman-release.spec [hunk 2]
```diff
@@ -66,6 +79,19 @@ install -Dpm0644 %{SOURCE1} %{buildroot}%{repo_dir}/foreman-plugins.repo
 install -Dpm0644 %{SOURCE3} %{buildroot}%{repo_dir}/foreman-rails.repo
 install -Dpm0644 %{SOURCE5} %{buildroot}%{repo_dir}/foreman-client.repo
 
+%if 0%{?rhel}
+%endif
+
+%if 0%{?rhel} == 6
+install -m 644 %{SOURCE6} %{buildroot}%{repo_dir}/pulp.repo
```
Comment (ekohl) [inline packages/foreman/foreman-release/foreman-release.spec]: SOURCE8 here

File: packages/foreman/foreman-release/foreman-release.spec [hunk 3]
```diff
@@ -66,6 +79,18 @@ install -Dpm0644 %{SOURCE1} %{buildroot}%{repo_dir}/foreman-plugins.repo
 install -Dpm0644 %{SOURCE3} %{buildroot}%{repo_dir}/foreman-rails.repo
 install -Dpm0644 %{SOURCE5} %{buildroot}%{repo_dir}/foreman-client.repo
 
+%if 0%{?rhel}
```
Comment (ekohl) [inline packages/foreman/foreman-release/foreman-release.spec]: This feels like a dummy

**Comment by ehelms** [comment]
@ekohl Good with this update?

File: package_manifest.yaml [hunk 4]
```diff
@@ -398,6 +398,7 @@ foreman_client_packages:
       - el6-puppet-5
       - el6-subscription-manager
```
Comment (ekohl) [inline package_manifest.yaml]: This doesn't sound correct if I look at the repos here

Comment (ehelms) [reply package_manifest.yaml]: What do you mean?

Comment (ekohl) [reply package_manifest.yaml]: In the client repos we don't install the subscription manager copr for EL6 so why is it needed for this repoclosure?

**Pull request merged**